http_server/request_testing/request_test.go

```diff
@@ -0,0 +1,112 @@
+package requesttesting
+
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"testing"
+)
+
+type handlerRecorder struct {
+	req         *http.Request
+	finishedReq chan bool
+	//TODO(mara): add server response
+}
+
+
+func (hr *handlerRecorder) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	hr.req = r
+	log.Print("Handler was called.")
+	// log.Print(hr.req.URL.Query()["vegetable"])
+	hr.finishedReq <- true
+}
+
+func makeRequest(req string) *http.Request {
+	e1, e2 := net.Pipe()
+	listener := newMockListener(e2)
+	done := make(chan bool)
+	defer listener.Close()
+	handler := &handlerRecorder{req: nil, finishedReq: done}
```

Comment:
you can use the `make` call inside the struct literal, you don't need to have a separate variable for it. Try to restrict the scope of your variable to the semantic block they live in (e.g. in this case you never use "done" again, so it is better to hide it, or a reader of the code might be wondering why you're saving this).